### PR TITLE
[WFCORE-867] Delegate to the AbstractAddStepHandler to discover if a runtime step is required.

### PR DESCRIPTION
**Pull request by darranl**

### CODE REVIEW

**Comment by darranl** [comment]
This needs to be tested again after the following PR has been merged - https://github.com/wildfly/wildfly/pull/7877

For some reason ServiceRemoveStepHandler was extended and most of it's methods overidden.

**Comment by darranl** [comment]
Retest this please

File: controller/src/main/java/org/jboss/as/controller/ServiceRemoveStepHandler.java [hunk 1]
```diff
@@ -1,3 +1,25 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2013, Red Hat, Inc., and individual contributors
```
Comment (jamezp) [inline controller/src/main/java/org/jboss/as/controller/ServiceRemoveStepHandler.java]: If we're adding a copyright we might as well use the current year :)

**Comment by jamezp** [comment]
@bstansberry WDYT?

File: controller/src/main/java/org/jboss/as/controller/ServiceRemoveStepHandler.java (hunk 1 above)
Comment (darranl) [inline controller/src/main/java/org/jboss/as/controller/ServiceRemoveStepHandler.java]: No this is the year the class was written - 2013 is not an accident ;-)

**Comment by bstansberry** [comment]
@jamezp it's fine. It's a behavior change but a totally logical one and WF 10 is a major.

When I saw this JIRA this morning I was quite nervous about scope, but this particular case is quite limited in scope.

**Pull request merged**